sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1000,12 +1000,22 @@ def is_cmis_application_update_required(self, api, app_new, host_lanes_mask):
         for lane in range(self.CMIS_MAX_HOST_LANES):
             if ((1 << lane) & host_lanes_mask) == 0:
                 continue
+            app_cur = api.get_application(lane)
             if app_old == 0:
-                app_old = api.get_application(lane)
-            elif app_old != api.get_application(lane):
+                app_old = app_cur
+            elif app_cur != 0:
```

Comment:
@AnoopKamath Please test the following
Have one subport in shutdown state as part of startup config. Reboot the device and allow the neighboring subport to be linked up.
Now issue startup on the subport which was originally down and ensure that the datapath of the neighboring subport is not impacted and both the subports are linked up.

Reply:
Tested and made sure data path of neighboring subport is not impacted